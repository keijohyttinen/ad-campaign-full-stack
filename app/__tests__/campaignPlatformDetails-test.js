import 'react-native';
import React from 'react';
import fs from 'fs';
import CampaignPlatformDetails from '../src/components/campaignPlatformDetails';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders CampaignPlatformDetails correctly', () => {
  let data = JSON.parse(fs.readFileSync('__tests__/resources/data.json'));
  const props = {
    name: "Campaing Row prints TestName-2",
    navigation: {
      navigate: () => { },
      state: {
        name: "Name1",
        params: {
          data: Object.assign({ "platformName": "Facebook" }, data[0].platforms.facebook)
        }
      },
    },
  };
  const tree = renderer.create(<CampaignPlatformDetails {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
