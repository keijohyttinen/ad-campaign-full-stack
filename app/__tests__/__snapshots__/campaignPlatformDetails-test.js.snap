// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CampaignPlatformDetails correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Your Ad
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "height": 1,
                },
                Array [
                  Object {
                    "marginBottom": 15,
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 600,
              "opacity": 1,
              "width": 750,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flex": 2,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 93.75,
                  "height": 187.5,
                  "width": 625,
                },
              ]
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "undefined:undefined/images/img1.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 5,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                },
                Object {
                  "flex": 1,
                  "marginBottom": 10,
                  "width": 625,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "fontSize": 27.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "marginBottom": 5,
                    },
                    Object {
                      "flexDirection": "column",
                      "flexWrap": "wrap",
                      "fontSize": 16,
                      "width": 625,
                    },
                  ],
                ]
              }
            >
              DevOps Made Easy, We Take care of the heavy lifting for you
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "flexWrap": "wrap",
                  "fontSize": 12,
                  "width": 535.7142857142858,
                }
              }
            >
              <Text>
                DOP SuperHero is where all DevOps is going to happen in the future, join the revolution today!
              </Text>
              <Text
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                https://example.io
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Target Audience
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "height": 1,
                },
                Array [
                  Object {
                    "marginBottom": 15,
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Languages
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="FR,EN,DE"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Genders
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="M,F"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Age Range
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="20 - 66"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Locations
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="France,Germany,Switzerland"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Interests
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="Docker,Kubernates,DevOps,AWS,Google Cloud Platform,Ubuntu"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Insight
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "height": 1,
                },
                Array [
                  Object {
                    "marginBottom": 15,
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Impressions
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="4503"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Clicks
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="328"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Score
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="5.7"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Cost/Click
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="0.88"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Click Through Rate
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="0.09"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Advanced KPI
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="44.5"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              Advanced KPI2
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              style={
                Object {
                  "marginBottom": 5,
                }
              }
              underlineColorAndroid="transparent"
              value="0.0023"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
